src/text_generator.py

```diff
@@ -0,0 +1,132 @@
+                    services = ", ".join(not_subscribed_services[0:3])
+                else:
+                    services = ", ".join(not_subscribed_services)
+                text = text.replace('{$proposeNewService}',
+                                    'It seems, you are very interested in service ' + subscribed_services
+                                    [0] + '. You should also check these services: ' + services + '!')
+            else:
+                if len(not_subscribed_services) >= 3:
+                    text = text.replace('{$proposeNewService}',
+                                        'Looking for new action? Check our best services: ' + ', '.join(
+                                            not_subscribed_services[0:3]) + '!')
+                else:
+                    text = text.replace('{$proposeNewService}',
+                                        'Looking for new action? Check our best services: ' +
+                                        ', '.join(not_subscribed_services) + '!')
+        else:
+            text = text.replace('{$proposeNewService}',
+                                'It seems, you are one of our best customers! You\'ve bought all of our services! '
+                                'Stay tuned for more!')
+
+    if '{$proposeLengtheningSubscription}' in text:
+        subscribed_services = database.get_subscribed_services()
+        if len(subscribed_services) > 0:
+            text = text.replace('{$proposeLengtheningSubscription}',
+                                'The subscription for ' + subscribed_services[0] + ' will soon expire! Quick! Renew the '
+                                                                                   'subscription!')
+        else:
+            # delete mark, because there is nothing to propose
+            text = text.replace('{$proposeLengtheningSubscription}',
+                                '')
+
+    if '{$suggestContact}' in text:
+        text = text.replace('{$suggestContact}',
+                            'We are glad, you are interested in our services. For more information, we suggest '
+                            'visiting our website to check new products we offer!')
+
+    if '{$goodbye}' in text:
+        text = text.replace('{$goodbye}',
+                            'Enjoy our services \nDC Drying Paint Services')
+
+    if '{$subscribedServices}' in text:
+        text = text.replace('{$subscribedServices}',
+                            ', '.join(database.get_subscribed_services()))
+
+    if '{$notSubscribedServices}' in text:
+        text = text.replace('{$notSubscribedServices}', ', '.join(database.get_not_subscribed_services()))
+
+    return text
+
+
+def get_propose_mail_text(userID: int, database) -> str:
+    """
+    function for creating mail text procedurally
+    :param userID:
+    :param database: the database object - at the moment mock object
+    :return: mail text
+    """
+    mail_text = ('{$greeting}\n'
```

Comment:
Do you mean the input text becomes a list? I've just didn't have any more idea what text could i generate. Even though  f.e. {$goodbye} is constant text, i left it this way, because somoene might get an idea, how can it be different for different customers. The invoice mail has more unparametrized text as input.